### DISCUSSION TITLE
Proposal for `parse_tsv_matrix` refactoring

### DISCUSSION THREAD

**Discussion by laurent-laporte-pro** [AntaresSimulatorTeam/AntaREST#1250] [Show and tell]
The `parse_tsv_matrix` could be efficiently refactored like that:

```python
def parse_tsv_matrix(file_data: bytes) -> List[List[MatrixData]]:
    """
    Parse a TSV file data and return a matrix of data.
    
    Args:
        file_data : The TSV file data
    
    Returns:
        The matrix of data.
    """
    file = io.BytesIO(file_data)
    reader = csv.reader(file, delimiter="\t")
    return [[MatrixData(elm) for elm in row] for row in filter(None, reader)]
```

**Comment by sylvlecl** [comment]
Actually, using pandas to read those CSVs will be a LOT more efficient than a for loop.
It means changing the return type, though, so it has some impact. A plain `List[List[MatrixData]]` is a very poor representation, in python, for a numeric matrix, especially for the typical 8760x200 matrices that we use!

It's already done this way in raw study services, it seems: not sure why it's not homogeneous ?
See here:
https://github.com/AntaresSimulatorTeam/AntaREST/blob/d6eae5342fd1f731bbc60392428a9e0529e1de32/antarest/study/storage/rawstudy/model/filesystem/matrix/input_series_matrix.py#L40-L45
